src/main/scala/cognite/spark/v1/CdpConnector.scala

```diff
@@ -80,12 +80,22 @@ object CdpConnector {
         case _ => IO.unit
       })
 
+  def throttlingSttpBackend(sttpBackend: SttpBackend[IO, Any]): SttpBackend[IO, Any] = {
+    // this backend throttles all requests when rate limiting from the service is encountered
+    val makeQueueOf1 = for {
+      queue <- Queue.bounded[IO, Unit](1)
+      _ <- queue.offer(())
+    } yield queue
+    new BackpressureThrottleBackend[IO, Any](sttpBackend, makeQueueOf1.unsafeRunSync(), 800.milliseconds)
```

Comment:
I know this was here before, but won't this block the thread every time more than one request is queued for this backend? Isn't there an async alternative for this?

Reply:
You mean the wait on queue take/offer?
https://github.com/cognitedata/cognite-sdk-scala/blob/master/src/main/scala/com/cognite/sdk/scala/sttp/BackpressureThrottleBackend.scala#L30 uses it as a mutex
On no 503/429 it is quickly acquired and released, but on 503/429 the request hitting those takes it for longer and blocks other requests by design.
This should all be within cats-effects mechanism so shouldn't block on compute threads but rather shift to io threads or do async suspend.